presto-hive/src/main/java/com/facebook/presto/hive/EncryptionInformation.java

```diff
@@ -77,4 +82,12 @@ public boolean equals(Object obj)
         EncryptionInformation otherObj = (EncryptionInformation) obj;
         return Objects.equals(this.dwrfEncryptionMetadata, otherObj.dwrfEncryptionMetadata);
     }
+
+    public Map<Integer, Slice> toKeyMap(List<OrcType> types, List<HiveColumnHandle> physicalColumnHandles)
+    {
+        if (dwrfEncryptionMetadata.isPresent()) {
+            return dwrfEncryptionMetadata.get().toKeyMap(types, physicalColumnHandles);
+        }
+        return ImmutableMap.of();
+    }
```

Comment:
do other file formats have a similar format where in column values are stored as integers? If not, we should just let the Orc code do `getDwrfEncryptionMetadata` and then call `toKeyMap`.

My idea behind this class was to allow for future encryption implementations to be dropped into this class.

Reply:
Hmmm... it's definitely true for Orc and DWRF.  Parquet has some notion of column index, but it's not clear whether the index or name is more commonly used for accessing columns.

I can get rid of this and change OrcBatchPageSourceFactory and OrcSelectivePageSourceFactory to the following:

```
Map<Integer, Slice> keyMap = ImmutableMap.of();
if (encryptionInformation.isPresent() and encryptionInformation.getDwrfEncryptionMetadata.isPresent()
{
    keyMap = encryptionInformation.getDwrfEncryptionInformation.toKeyMap();
}
```